supervisor/store/repository.py

```diff
@@ -12,36 +15,32 @@
 from ..exceptions import ConfigurationFileError, StoreError
 from ..utils.common import read_json_or_yaml_file
 from .const import StoreType
-from .git import GitRepo, GitRepoBuiltin, GitRepoCustom
+from .git import GitRepo
 from .utils import get_hash_from_repository
 from .validate import SCHEMA_REPOSITORY_CONFIG, BuiltinRepository
 
 _LOGGER: logging.Logger = logging.getLogger(__name__)
 UNKNOWN = "unknown"
 
 
-class Repository(CoreSysAttributes):
+class Repository(CoreSysAttributes, ABC):
     """Add-on store repository in Supervisor."""
 
     def __init__(self, coresys: CoreSys, repository: str):
         """Initialize add-on store repository object."""
+        self._slug: str
+        self._type: StoreType
```

Comment:
I wonder if we can maybe get rid of this type attribute entirely :thinking: . This should be done in a follow up PR as it likely has some churn on its own.